From 1d7adb9a82e6924d3ab47f3bf8b245dfe85b1e6d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:15 -0400
Subject: [powerpc] Make add_system_ram_resources() __init

Message-id: <1410545655-205645-267-git-send-email-dzickus@redhat.com>
Patchwork-id: 94219
O-Subject: [RHEL7 PATCH 266/626] powerpc: Make add_system_ram_resources() __init
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 4f7709248db7402507a2a5164f86ad4c75a1926b
Author: Geert Uytterhoeven <geert@linux-m68k.org>
Date:   Sun Sep 15 11:39:36 2013 +0200

    powerpc: Make add_system_ram_resources() __init

    add_system_ram_resources() is a subsys_initcall.

    Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/mem.c b/arch/powerpc/mm/mem.c
index d678d47..9dbe09e 100644
--- a/arch/powerpc/mm/mem.c
+++ b/arch/powerpc/mm/mem.c
@@ -554,7 +554,7 @@ void update_mmu_cache(struct vm_area_struct *vma, unsigned long address,
  * System memory should not be in /proc/iomem but various tools expect it
  * (eg kdump).
  */
-static int add_system_ram_resources(void)
+static int __init add_system_ram_resources(void)
 {
  struct memblock_region *reg;
 
-- 
1.7.1